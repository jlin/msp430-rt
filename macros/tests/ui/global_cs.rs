#![no_main]

use msp430::interrupt::CriticalSection;
use msp430_rt_macros::{entry, interrupt};

static mut CS: Option<CriticalSection> = None;

#[entry]
fn main(cs: CriticalSection) -> ! {
    unsafe {
        CS = Some(cs);
    }
    loop {}
}

#[interrupt]
fn DefaultHandler(cs: CriticalSection) -> ! {
    unsafe {
        CS = Some(cs);
    }
    loop {}
}
