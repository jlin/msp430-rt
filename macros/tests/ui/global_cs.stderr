error[E0658]: msp430-interrupt ABI is experimental and subject to change
  --> tests/ui/global_cs.rs:16:1
   |
16 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: see issue #38487 <https://github.com/rust-lang/rust/issues/38487> for more information
   = help: add `#![feature(abi_msp430_interrupt)]` to the crate attributes to enable
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0759]: `cs` has lifetime `'a` but it needs to satisfy a `'static` lifetime requirement
  --> tests/ui/global_cs.rs:11:14
   |
8  | #[entry]
   | -------- this data with lifetime `'a`...
...
11 |         CS = Some(cs);
   |              ^^^^^--^
   |                   |
   |                   ...is used and required to live as long as `'static` here

error[E0570]: `"msp430-interrupt"` is not a supported ABI for the current target
  --> tests/ui/global_cs.rs:16:1
   |
16 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0759]: `cs` has lifetime `'a` but it needs to satisfy a `'static` lifetime requirement
  --> tests/ui/global_cs.rs:19:14
   |
16 | #[interrupt]
   | ------------ this data with lifetime `'a`...
...
19 |         CS = Some(cs);
   |              ^^^^^--^
   |                   |
   |                   ...is used and required to live as long as `'static` here
